src/callbacks_step/glm_speed.jl

```diff
@@ -65,19 +74,59 @@ function (glm_speed_callback::GlmSpeedCallback)(u, t, integrator)
     return true
 end
 
-# This method is called as callback after the StepsizeCallback during the time integration.
-@inline function (glm_speed_callback::GlmSpeedCallback)(integrator)
-    dt = get_proposed_dt(integrator)
-    semi = integrator.p
-    mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+function update_cleaning_speed!(semi, glm_speed_callback, dt)
     @unpack glm_scale, cfl = glm_speed_callback
 
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
     # compute time step for GLM linear advection equation with c_h=1 (redone due to the possible AMR)
     c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
 
     # c_h is proportional to its own time step divided by the complete MHD time step
     equations.c_h = glm_scale * c_h_deltat / dt
 
+    return semi
+end
+
+function update_cleaning_speed!(semi_coupled::SemidiscretizationCoupled,
+                                glm_speed_callback, dt)
+    @unpack glm_scale, cfl, semi_indices = glm_speed_callback
+
+    if length(semi_indices) == 0
+        throw("Since you have more than one semidiscretization you need to specify the 'semi_indices' for which the GLM speed needs to be calculated.")
+    end
+
+    # Check that all MHD semidiscretizations received a GLM cleaning speed update.
+    for (semi_index, semi) in enumerate(semi_coupled.semis)
+        if (typeof(semi.equations) <: AbstractIdealGlmMhdEquations &&
+            !(semi_index in semi_indices))
+            error("Equation of semidiscretization $semi_index needs to be included in 'semi_indices' of 'GlmSpeedCallback'.")
+        end
+    end
+
+    for semi_index in semi_indices
+        semi = semi_coupled.semis[semi_index]
+        mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+        # compute time step for GLM linear advection equation with c_h=1 (redone due to the possible AMR)
+        c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
+
+        # c_h is proportional to its own time step divided by the complete MHD time step
+        equations.c_h = glm_scale * c_h_deltat / dt
+    end
+
+    return semi_coupled
+end
```

Comment:
```suggestion
```
Delete this since you moved it to `src/semidiscretization/semidiscretization_coupled.jl`?